app/access/api.py

```diff
@@ -101,3 +104,34 @@ def delete(request: HttpRequest, username: str) -> HttpResponse:
             raise HttpError(500, "Internal Server Error")
         user_to_delete.delete()
         return HttpResponse(status=204)
+
+
+@router.put("users/{username}")
+def update_user(
+    request: HttpRequest, username: str, user_in: UserSchema
+) -> HttpResponse | UserSchema:
+    """Update the given user with the given user data and return it.
+
+    Return HTTP status code
+    - 200 (OK) if the user has been updated successfully
+    - 404 (Not Found) if there is no user with the given username or if there is
+      no provider with the given ID
+    - 500 (Internal Server Error) if there is an inconsistency with Cognito
+    - 503 (Service Unavailable) if Cognito cannot be reached
+    """
+    with transaction.atomic():
+        user_object = User.objects.select_for_update().filter(username=username).first()
+        if not user_object:
+            raise Http404()
+
+        if not Provider.objects.filter(id=user_in.provider_id).exists():
+            raise HttpError(HTTPStatus.NOT_FOUND, "Provider does not exist")
```

Comment:
was mentioned previously: should be `400`

Reply:
Ah yes, thanks for the reminder. Fixed!